spacemacs/packages.el

```diff
@@ -2492,7 +2492,7 @@ displayed in the mode-line.")
 
 (defun spacemacs/init-volatile-highlights ()
   (use-package volatile-highlights
-    :init
+    :config
```

Comment:
So if `:config` is an eval-after-load it make sense to me that you need to
require it manually, you have no trigger to launch the auto load. We need a
binding (not relevant here) or a hook. Since this mode is useful globally I
don't see how to defer it.
I don't want loading on idle by the way. I don't who thought it would be
cool to have this feature, it has not practical use, things have to be
deterministic to be usable in a config.

Le samedi 14 mars 2015, Chris Barrett <notifications@github.com> a écrit :

> In spacemacs/packages.el
> <https://github.com/syl20bnr/spacemacs/pull/827#discussion_r26442411>:
>
> > @@ -2492,7 +2492,7 @@ displayed in the mode-line.")
> >
> >  (defun spacemacs/init-volatile-highlights ()
> >    (use-package volatile-highlights
> > -    :init
> > +    :config
>
> My intuition was close I think. Given the following:
>
> (use-package foo
>   :init
>   'foo-init
>   :config
>   'foo-config)
>
> We get the following macro-expansion (tidied up a bit for clarity)
>
> (progn
>   'foo-init
>   (if (not (require 'foo nil t))
>       (message "Could not load package %s" "foo")
>     'foo-config))
>
> Somehow I ran into a corner case when debugging my startup process where
> this package in particular wasn't getting autoloaded, so the call to
> volatile-highlights-mode would throw an error. Aggressively requiring the
> feature fixes this.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/syl20bnr/spacemacs/pull/827/files#r26442411>.
>


-- 
-syl20bnr-